src/library/librarycontrol.cpp

```diff
@@ -1059,30 +1076,17 @@ void LibraryControl::slotDecrementFontSize(double v) {
     }
 }
 
-void LibraryControl::slotTrackColorPrev(double v) {
-    if (!m_pLibraryWidget) {
+void LibraryControl::slotTrackColorSelector(int steps) {
+    if (!m_pLibraryWidget || steps == 0) {
         return;
     }
 
-    if (v > 0) {
-        LibraryView* pActiveView = m_pLibraryWidget->getActiveView();
-        if (!pActiveView) {
-            return;
-        }
-        pActiveView->assignPreviousTrackColor();
-    }
-}
-
-void LibraryControl::slotTrackColorNext(double v) {
-    if (!m_pLibraryWidget) {
+    LibraryView* pActiveView = m_pLibraryWidget->getActiveView();
+    if (!pActiveView) {
```

Comment:
FWIW I have a WIP branch that makes use of [`WLibrary::getCurrentTrackTableView()`](https://github.com/mixxxdj/mixxx/blob/9b33757a4bc1dd38a3c2ab837c20c57bb771875d/src/widget/wlibrary.cpp#L105). This allows to remove many virtual methods from libraryview.h and library features (DlgMissing etc.) which essentially just redirect calls to their own WTrackTableView (which you didn't do here, which means the color controls don't work in AutoDJ, Missing and Hidden?).

Reply:
I'm sorry, my knowledge of the codebase is clearly insufficient to understand this comment.

I cannot find the `getCurrentTrackTableView` method anywhere.
Why would AutoDJ need these features ?
How does he access them ?

Can you clarify or point me to the relevant files ?

Reply:
sorry, I was knee-deep into it when I wrote this.

Like the color select slots, many control slots in LibraryControl call methods of the 'active view'. This can be anything that inherits LibraryView: a pure WTrackTableView, a html root view (top-level item of Crates and Playlists), or some controls row + table view (Auto DJ, Recording, Analysis).

So, until now, a virtual function in libraryview.h needs to be overriden not just in WTrackTableView but in every view/feature where a certain control should be availble. "Until now" because I opened #13335 earlier today which makes use of [WLibrary::getCurrentTrackTableView()](https://github.com/mixxxdj/mixxx/blob/9b33757a4bc1dd38a3c2ab837c20c57bb771875d/src/widget/wlibrary.cpp#L105) to get rid of all the virtuals and overrides (see what I removed there).
That's why AutoDJ doesn't **need** it, but you probably want the color controls to work in the AutoDJ and Recording table, too.

> I cannot find the `getCurrentTrackTableView` method anywhere.

Probably because your branch is based on an older version of main that doesn't have it, yet, not sure.

Reply:
FWIW I've been using the Github code search a ot lately, very handy:
https://github.com/search?q=repo%3Amixxxdj%2Fmixxx%20%22searchterm%22&type=code
I have a search accelerator (shortcut) and I wrap the term in quotes " (%22) so I use search terms with whitespaces.